src/array.cr

```diff
@@ -590,6 +590,51 @@ class Array(T)
     elem
   end
 
+  # Deletes all elements that are within the given range,
+  # returning that elements.
+  # Raises `IndexError` if the index is out of range.
+  #
+  # ```
+  # a = ["ant", "bat", "cat", "dog"]
+  # a.delete_at(1..2)  #=> ["bat", "cat"]
+  # a                  #=> ["ant", "dog"]
+  # a.delete_at(99..100)  #=> IndexError
+  # ```
+  def delete_at(range : Range)
+    return nil if range.first < 0 && range.end > 0
```

Comment:
Same as the last comment, spec & probably should be empty array. Unions where we don't really need one are always a bit annoying to deal with.